mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -41,6 +42,7 @@ struct MiirHandle_s {
     auto getRegistry = []() {
       DialectRegistry registry;
       registerAllDialects(registry);
+      registerMIOpenDialects(registry);
```

Comment:
>  LLVM ERROR: Building op `func.func` but it isn't registered in this MLIRContext: the dialect may not be loaded or this operation isn't registered by the dialect. See also https://mlir.llvm.org/getting_started/Faq/#registered-loaded-dependent-whats-up-with-dialects-management
> ...
> [2022-07-14T18:51:28.912Z] 0.	Program arguments: mlir-rocm-runner --shared-libs=/var/z1_miciadmin/workspace/MLIR_mlir_PR-636/build/external/llvm-project/llvm/lib/libmlir_rocm_runtime.so,/var/z1_miciadmin/workspace/MLIR_mlir_PR-636/build/lib/libconv-validation-wrappers.so,/var/z1_miciadmin/workspace/MLIR_mlir_PR-636/build/external/llvm-project/llvm/lib/libmlir_runner_utils.so --entry-point-result=void

I think according to the stack trace it has less to do with `libMLIRMIOpen` but more to do with `mlir-rocm-runner`.

Maybe try to add a `context.loadDialect<func::FuncDialect>` into `mlir-rocm-runner` to make sure it is loaded.

Reply:
I'd think the error is a consequence of the func not getting lowered to llvm properly

But I wouldn't be seeing the issue if I'd marked these things `kernel`??

Reply:
> I'd think the error is a consequence of the func not getting lowered to llvm properly

Agreed, rocm runner shouldn't expect a func op.

> But I wouldn't be seeing the issue if I'd marked these things kernel??

We picked those func in `MIOpen -> GPU` pass that has `kernel` attributes. Sounds like the way to go?